### ISSUE TITLE
New Feature: Work In Progress Limits

### ISSUE THREAD

**Issue by bewalterBL** [edtro/EdTro.AzureDevOps.Extensions.querybasedboards#30]
Hey Edward,

it would be great, if you could introduce WIP limits similar to ones offered on the original boards. Maybe it would even be possible to have seperate WIP limits for each swimlane.
And now the killer feature: Maybe the WIP limit could be enforced in a way that if the wip limit of a column is reached, it is not allowed to drag an item to it.

Best Bernhard

**Comment by edtro** [comment]
Hi Bernhard,
Yes indeed, this is something that I should add.
But however... I don't know if a WIP limit for each swimlane how this should work, I would have to look into that.

But blocking on a limit... I do not see the added value directly, because within the card you can also set the state, so there is an very easy way around that blocker. Next to that... are you sure that you want to do this? I think that your teammembers or projectmanager will not agree with blocking on this. 

I will be looking into this... but I cannot give you any timetable, about when or even if I am going to implement it.

But please, if you got any ideas, please do not hesitate to post them, I will look into them and these will all be considered.
(but because I am just alone and doing this is in my free spare time, I am very limited regarding what I can pick up)

Kr.
Edward

**Comment by bewalterBL** [comment]
Yes, you are right. The blocking doesn't make sense when there is such an easy workaround. It just came to my mind because at some time our numbers were constantly red :-).

But even the wip limits as they are in the standard boards would be a great benefit.

Thank you so much for the quick reply.
Best Bernhard

**Comment by merleschlottmann** [comment]
Hi Edward, 
I really like your QuerybasedBoards as well and would really much appreciate a sort of numbering as well so that we can keep track of the WIP items. It would not need to be a strong limit, as you say that might be even annoying. 
But I could image different solutions: 
-  only count the number of items in a column and we know about our internal limit 
- similar to the general backlog board where you could add a WIP limit and the count of items will turn red if exceeded

Best,
Merle

**Comment by edtro** [comment]
Hi, I am working on it... here is a preview screenshot:
![image](https://user-images.githubusercontent.com/27762373/102640051-319a7400-415a-11eb-8bc8-cef1f337b344.png)

You will be able to setup the wiplimits within the config json.

**Comment by bewalterBL** [comment]
That's great to hear! Thank you for approaching this so quickly.

Would it also be possible to have inverse WIP limits?  So that the numbers turn red when the number of items falls below the limit. I am thinking about Scrumban, where the items in a column dropping below a certain number would trigger a planning meeting. But it would be necessary to have this adjustable per column.

I whish you happy holidays!!! 
Best Bernhard

**Comment by edtro** [comment]
Hi Bernhard,

To be honest, I never heard of this approach before.... but I will take a look at it. But I am in the middle of a new release at the moment and I am not including any new things for this release. Next I will be picking up a completely other item... and this will be a challenging one regarding displaying everything correctly... thats all I want to say about this item, at this time :-) 

So stay tuned.

Kr
Edward

**Comment by edtro** [comment]
@bewalterBL  
btw. I also wish you happy holidays & stay safe!

**Comment by edtro** [comment]
I am happy to inform you, that I have released this new feature. Because it is a part of setting up the columns within the config... this will be in PREVIEW

**Comment by edtro** [comment]
whoops closed this issue... while there was still a suggestion open to create 'inverse'  wip limits for a 'scrumban' approach.

**Comment by bewalterBL** [comment]
Thank you very much for implementing this so quickly!!! I tried it out and it works perfectly.

**Comment by merleschlottmann** [comment]
Thanks also from my side. That is super helpful extension.

**Comment by Wandernoth** [comment]
hi @edtro,

Is there a reason, this is still in preview and not GA?

**Comment by edtro** [comment]
@Wandernoth it actually is GA.
My appologies for not updating the tickets.
Kr
Edward

**Comment by Wandernoth** [comment]
Huh, interesting.

I was hoping that this was the reason, the WIP Limits do not appear for me. 
I am using Version: 0.23028.2 and the following Json for configuration of a one-hop query taskboard:
```
{
   "columns":[
      {
         "name":"New",
         "title":"New/Backlog",
         "isBacklog":true
      },
      {
         "name":"Active",
         "title":"In Progress/Active",
         "wipLimit":2
      },
      {
         "name":"Resolved",
         "title":"Resolved",
         "wipLimit":10
      },
      {
         "name":"Waiting",
         "title":"Waiting"
      },
      {
         "name":"Closed",
         "title":"Closed",
         "isBacklog":true
      }
   ]
}
```
and the corresponding header of the board looks like this:
![image](https://user-images.githubusercontent.com/15245173/215544483-c6976423-a700-4710-87e3-02d3306e89b2.png)

Do you have any idea, why it doesn't show up?

Thanks in advance,
Sebastian
